### PR TITLE
[MM-454] Fix Invalid message while creating subscription for a project with no access to create webhook

### LINKED ISSUE #454: Invalid message while creating subscription for someone else's project
#### Summary
When a user tries to create a subscription for a project he/she is not an owner of, he/she gets an invalid response. Also, the link attached in the message is not correct.
This message is basically because we are unable to fetch the webhook details for that project. We can still create a subscription if we comment out this part of the code.
https://github.com/mattermost/mattermost-plugin-gitlab/blob/7834d0fc36f08badf9d6827c8e0cca741e56f6d5/server/command.go#L632

#### Screenshot:
image

#### Steps to reproduce:
1. Connect your Gitlab account.
2. Try creating a subscription for a project you are not an owner of.

### LINKED ISSUE #509: Update documentation link for webhook setup
#### Summary
When the webhook is not setup for a project we have created subscription for. We get a response containing the link for webhook setup. That link is pointing to older readme file.

Permalink:
https://github.com/mattermost/mattermost-plugin-gitlab/blob/01b5e6c2842f4df4271866ef6c089b9278f7c5f0/server/command.go#L65

### PR DESCRIPTION
**Pull request by raghavaggarwal2308**
#### Summary
- Users we not able to create subscription for a project in which they did not had access to create webhook.
- Updated the code to post a message in the above case instead preventing the user from creating subscription.

#### Screenshots:
image

#### What to test?
- Getting proper messages while creating subscription  for a project with no access to create webhook.

#### How to test?
1. Connect your Gitlab account
2. Try to create a subscrition for a project with no access to create subscription

#### Ticket Link
Fixes #454
Fixes #509

### CODE REVIEW

File: server/command.go [hunk 1]
```diff
@@ -632,31 +631,27 @@ func (p *Plugin) subscriptionsAddCommand(ctx context.Context, info *gitlab.UserI
...
 	var hasHook bool
+	hookErrorMessage := ""
 	if project != "" {
 		hasHook, err = p.HasProjectHook(ctx, info, namespace, project)
 		if err != nil {
-			return fmt.Sprintf(
-				"Unable to determine status of Webhook. See [setup instructions](%s) to validate.",
-				webhookHowToURL,
-			)
+			hookErrorMessage = "\n**Note:** We are unable to determine the webhook status for this project. Please contact your project administrator"
```
Comment (ayusht2810) [inline server/command.go:640]: we can create constant for this error

File: server/command.go [hunk 2]
```diff
@@ -632,31 +631,27 @@ func (p *Plugin) subscriptionsAddCommand(ctx context.Context, info *gitlab.UserI
...
 	}
+
 	var hookStatusMessage string
 	if !hasHook {
 		// no web hook found
-		hookStatusMessage = fmt.Sprintf(
-			"\nA Webhook is needed, run ```/gitlab webhook add %s``` to create one now.",
-			fullPath,
-		)
+		hookStatusMessage = fmt.Sprintf("\nA Webhook is needed, run ```/gitlab webhook add %s``` to create one now.%s", fullPath, hookErrorMessage)
```
Comment (ayusht2810) [inline server/command.go:654]: What if webhook was present but were unable to fetch due to some reasons? Is it fine to show this error?

Comment (raghavaggarwal2308) [reply server/command.go:654]: @ayusht2810 We are mentioning that we are unable to fetch the details. So, I think it is fine. Please let me know your opinions. 
Showing the webhook message is a additional feature only, the main task here is to create subscription.

**Review by wiggin77 — APPROVED** [review]
LGTM

**Review by ayusht2810 — APPROVED** [review]

**Review by arush-vashishtha — APPROVED** [review]
This PR has been tested and verified the message while creating subscription for a project in which the user has the role of a developer. LGTM.

**Pull request merged**